### ISSUE TITLE
@LogClass is conflicting with other decorators

### ISSUE THREAD

**Issue by phillippbertram** [aigoncharov/class-logger#6]
I just was trying out your logging decorators. They work nicely, but only if **no other class decorators** are used :/

Here is an example when using `@LogClass` in combination with `@injectable` from [InversifyJS](https://github.com/inversify/InversifyJS).

```typescript

@LogClass()
@injectable()
class MyService {

   constructor(@inject(OtherService) private otherService: OtherService) { ... }

   @Log()
   methodThatShouldBeLogged() { ... }

}
```

This will produce an error with following message:
```
Error: Missing required @injectable annotation in: MyService.
```

Changing the order of the decorators does also produces an error:
```
Error: Missing required @inject or @multiInject annotation in: argument 0 in class MyService.
```

I think the problem is, that the `@LogClass` implementation replaces the class declaration with the provided constructor function and the other decorators are removed.

Is there a workaround for this or has anyone an idea how to fix this?

Thanks in advance 🤗

**Comment by vickylance** [comment]
@phillippbertram  Did you try reversing the order of the decorators?

**Comment by phillippbertram** [comment]
Yes, as I have written in the description ☝️ 😉
I think the problem is the returned `Proxy` object of the `@LogClass` decorator, which will replace the class declaration and therefore looses the other decorators.

**Comment by phillippbertram** [comment]
Unfortunately I believe this repository is not maintained anymore. The last issue #2 wasn't answered since April '19 :/

**Comment by aigoncharov** [comment]
@phillippbertram I guess I just saw that issue as low priority. This one I'll try to fix ASAP.

**Comment by phillippbertram** [comment]
@aigoncharov yay thanks :)

**Comment by vickylance** [comment]
@aigoncharov  Issue no #7  is also related to this

**Comment by aigoncharov** [comment]
Guys, it turns out the issue is in `reflect-metadata`. Check out [the thread](https://github.com/rbuckton/reflect-metadata/issues/106). I'll push the ugly fix for it shortly. Basically, I'll have to copy metadata manually, which I don't really wanna do, but it seems we ran out of options.

**Comment by aigoncharov** [comment]
Should be fixed in 1.3.0. Could you try it and reopen the issue if necessary?
